secure/key-manager/src/tests.rs

```diff
@@ -135,71 +97,66 @@ impl Node {
     }
 }
 
+/// The trait below is a LibraInterface wrapper that exposes several additional methods to better
+/// test the internal state of a LibraInterface implementation (e.g., during end-to-end and
+/// integration tests).
+trait LibraInterfaceWrapper: LibraInterface {
+    /// Returns the discover set associated with the discovery set address.
+    fn retrieve_discovery_set(&self) -> Result<DiscoverySet, Error>;
+
+    /// Returns the libra block resource associated with the association address.
+    fn retrieve_libra_block_resource(&self) -> Result<LibraBlockResource, Error>;
+
+    /// Return (and clear) all transactions that have been submitted to this interface since the
+    /// last time this method was called.
+    fn take_all_transactions(&self) -> Vec<Transaction>;
+}
+
 #[derive(Clone)]
-struct TestLibraInterface {
+/// A mock libra interface implementation that stores a list of transactions submitted via the API,
+/// and a pointer to the LibraDB from which to process API requests.
+struct MockLibraInterface {
```

Comment:
I'd argue it is a test interface whereas a mock would imply that we are pulling strings from the back, wherein this is actually a feature full interface...

Reply:
I see what you're saying, but in this case I disagree because this struct isn't really feature-full -- it just maintains a list of transactions internally, without ever passing them to an executor, updating the db it talks too etc. Hence, why I thought Mock was more appropriate... I can change it back if you feel strongly though!